### PR TITLE
moonwatch: replace Firebase design with offline math engine + spec (v4.5.0)

### PR DESCRIPTION
**Pull request by MahtraDR**
## Summary

Replaces the current Firebase crowdsource design of `moonwatch.lic` with a self-contained **offline closed-form prediction engine** that self-corrects from the player's own observed events. No network calls, no shared database (`dr-scripts.firebaseio.com`), no `;moonwatch correct` moonbot, and no cold-start penalty -- a fresh launch is accurate immediately and every observed rise/set tightens it.

**This intentionally retires the Firebase crowdsource layer.** Confirmed by audit that `moonwatch.lic` is the *only* consumer of `dr-scripts.firebaseio.com` in dr-scripts (nothing else reads the shared `moon_data` or uses the `correct` path), so removing it here frees the instance to be shut down with no other breakage. The instance owner/maintainer has signed off on dropping it.

## What it does

- **DRTime** -- full Elanthian calendar (year/month/day/anlas/rois/season/named time-of-day) and sun rise/set from an empirical day-of-year table (cosine fallback), anchored to `CALENDAR_EPOCH`.
- **Moons** -- Bresenham tick prediction: computes the exact 60s server tick the game fires on instead of an average that is perpetually a little off, plus lunar phase. The moon periods are OLS-fit fractional periods that **cross-validate against the DR client's own hard-coded sidereal constants to ~0.05s**, so they are game-code constants -- identical on every instance.
- **MoonwatchOffsetManager** -- per-body self-correction; offsets persist in `CharSettings`, which Lich already scopes per `"#{XMLData.game}:#{XMLData.name}"`, so offsets are isolated per instance **and** per character with no extra bookkeeping.
- **MoonwatchLogger / ServerResetTracker / MoonwatchUI** -- opt-in CSV calibration logging, shutdown/restart + phase-shift tracking, and the status window + `UserVars` publishing.

## Instance-awareness (v4.5.0)

New `MoonwatchInstance` module. Because the periods are identical across instances and offsets are already per-instance-scoped by Lich, no per-instance constants are needed. On a non-Prime instance (Platinum/Fallen/Test) the script prints a one-time startup notice that the absolute phase self-calibrates from observed events; predictions converge after the first rise/set. (This deliberately supersedes the old Prime-only `unless XMLData.game == "DR"` refuse-and-exit guard, since the engine works on every instance.)

## Consumers unchanged

`combat-trainer`, `gate`, `autocontingency`, and `mm` read only `UserVars.moons['visible']`, per-moon `timer`/`pretty`, and `.values[0..2]` -- all populated with a compatible schema. Verified no consumer reads the old Firebase data or `$moon_offsets`.

## Tests

- **New `spec/moonwatch_spec.rb`: 123 examples, full method coverage** (every `def` has a matching `describe`), with adversarial/boundary cases on the calendar/sun/moon/phase math (day/year boundaries, midnight wrap, pre-epoch timestamps, fractional-period drift, tick rounding), the three self-correction branches (mismatch / tick-OK / tick-realign) and the offset auto-reset thresholds, CSV logging (dedup + outlier rejection), reset/phase-shift tracking, the UI, and **multi-instance offset isolation** (`DR`/`DRX`/`DRF`/`DRT`).
- **`test/test_harness.rb`**: adds settable `XMLData.game` / `server_time` / `name` accessors (with reset), following the existing `DRStats`/`DRRoom` pattern. Purely additive.
- Full dr-scripts suite green (**2212 examples**), rubocop clean, ASCII-only.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Updated Moonwatch to version 4.5.0 with more accurate tick-based sun and moon predictions.
  * Added lunar phase names, progress, and countdown information.
  * Added automatic calibration using observed rise and set events.
  * Added optional event logging, server reset detection, and alias synchronization.
  * Added support for multiple game instances with calibration notices.

* **Bug Fixes**
  * Improved prediction accuracy through persisted offset correction and tick-error handling.
  * Added safeguards for stale status indicators and duplicate log entries.

* **Tests**
  * Added comprehensive coverage for astronomical calculations, calibration, logging, resets, UI updates, and aliases.

### CODE REVIEW

**Comment by MahtraDR** [comment]
@coderabbitai review in full

File: moonwatch.lic [hunk 1]
```diff
@@ -1,233 +1,2477 @@
...
+        moon,
+        date[:day_of_year],
+        observed_phase,
+        computed[:index],
+        computed[:name],
+        computed[:phase_angle],
+        computed[:orbital_angle]
+      ].join(',')
+    end
+  end
```
[the remark this answers is not in the document]
Comment (MahtraDR) [reply moonwatch.lic:1827]: Fixed in b6a345c1.

Added an RFC4180-style `csv_field` helper (quotes the value and doubles embedded quotes only when it contains a comma, double-quote, or newline) and applied it to `observed_phase` only -- column order and the other fields' serialization are unchanged. Added a spec that logs `bright, waxing gibbous` and asserts the field is quoted and the row still parses to 8 columns.

File: spec/moonwatch_spec.rb [hunk 2]
```diff
@@ -0,0 +1,1158 @@
...
+    end
+
+    describe '#log_sun_event' do
+      let(:date) { DRTime.calculate_date(now) }
+
+      it 'writes a sun row with the drift column when enabled' do
+        logger.log_sun_event(true, now, date, 0, 100)
+        expect(File.exist?(sun_csv)).to be true
+        expect(File.readlines(sun_csv).first).to match(/expected_day_length,\s*|drift/)
+      end
```
[the remark this answers is not in the document]
Comment (MahtraDR) [reply spec/moonwatch_spec.rb:779]: Fixed in b6a345c1.

Replaced the loose alternation with an exact full-header equality assertion, so `expected_day_length,drift,sun_offset,predicted_seconds` is verified in sequence:

```ruby
expect(File.readlines(sun_csv).first.chomp).to eq(
  "server_time,event,day_of_year,season,last_rise,last_set," \
  "day_interval,day_length,night_length,expected_day_length," \
  "drift,sun_offset,predicted_seconds"
)
```

The existing `sun_csv` read is preserved.

[the remark this answers is not in the document]
Comment (MahtraDR) [reply spec/moonwatch_spec.rb:779]: (Also addressed the nitpick from the review body, since it has no inline thread.)

Dropped the redundant inner `unless` in the `SUN_DEBUG_KEYWORDS_PATTERN` branch. `SUN_RISE_COMBINED_PATTERN` and `SUN_SET_COMBINED_PATTERN` are earlier `when` clauses in the same `case`, so any line reaching the debug branch cannot have matched them -- the guard was always true. `MoonwatchMessaging.debug` now runs directly (same message and `$debug_mode_mm` argument). Fixed in b6a345c1.
